kotest-assertions/kotest-assertions-shared/src/jsMain/kotlin/io/kotest/assertions/eq/isDataClass.kt

```diff
@@ -0,0 +1,8 @@
+package io.kotest.assertions.eq
+
+/**
+ * Determine if the object's class is is a data class.
+ *
+ * Not available on this platform.
+ */
+actual fun Any?.isDataClass(): Boolean = false
```

Comment:
The aforementioned reflections interface has `un <T : Any> isDataClass(kclass: KClass<T>): Boolean` so you can invoke that with (myobj::class)